ecl/eclcc/NEWACTIVITY.rst

```diff
@@ -0,0 +1,537 @@
+- They provide something concrete to aim towards when implementing the feature.
+- They provide a set of milestones to show progress.
+- They can be used to check the implementation on the different engines.
+
+As part of the design discussion we also started creating a list of useful test cases (they follow below in the order 
+they were discussed).  The tests perform varying functions.  Some of the tests are checking the core functionality works 
+correctly, while others check unusual situations and that strange boundary cases are covered.  The tests are not 
+exhaustive, but they are a good starting point and new tests can be added as the implementation progresses.
+
+The following is the list of tests that should be created as part of implementing this activity:
+
+1. Compare with values extracted from a SORT.
+   Useful to check the implementation, but also to ensure we clearly define which results we are expecting.
+2. PARTITION to 1 range, also 0, and +infinity  (constant and dynamic)
+3. Empty dataset in.
+4. All input entries are duplicates.
+5. Dataset smaller than number of ranges.
+6. Input sorted and reverse sorted.
+7. Normal data with small number of entries.
+8. Duplicates which cause empty ranges.
+9. Random distribution of numbers without duplicates.
+10. Local and grouped cases.
+11. SKEW that fails.
+12. Test scoring functions.
+13. Testing different skews that work on the same dataset.
+14. An example that uses all the keywords.
+15. Examples that have extra fields and not (stable/non stable).
+16. Globally partitioned already/all partition points on a single node.
+17. Same dataset apply reordering/distribution and compare two.
+18. Calculate just the 5 and 95 centiles from a dataset.
+19. Check non constant number of splits (and also in a child query).
+20. Transform that does something interesting to the sort order.
+21. Check the counts are correct for grouped and local operations.
+22. Call in a child query with options that depend on the parent row (e.g., num partitions).
+23. Split points that fall in the middle of two items.
+24. No input rows and ,DEDUP specified.
+
+Ideally any test cases for features should be included in the runtime regression suite, which is found in the
+testing/regress directory in the github repository.  Tests that check invalid syntax should go in the compiler 
+regression suite (ecl/regress). Commit 
+https://github.com/ghalliday/HPCC-Platform/commit/d75e6b40e3503f851265670a27889d8adc73f645 contains the test cases so 
+far.  Note, the test examples in that commit do not yet cover all the cases above.  Before the final pull request for the
+feature is merged the list above should be revisited and the test suite extended to include any missing tests.
+
+In practice it may be easier to write the test cases in parallel with the implementing the parser �
+since that allows you to check their syntax.  Some of the examples in the commit were created before work was done
```

Comment:
'was done' is not that good a selection, even for a colloquial blog post. I would replace with commenced.